core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -1090,14 +1090,23 @@ aot_compile_op_memory_copy(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx)
     if (!(dst_addr = check_bulk_memory_overflow(comp_ctx, func_ctx, dst, len)))
         return false;
 
+    if (comp_ctx->pointer_size == sizeof(uint64)) {
+        /* zero extend to uint64 if the target is 64-bit */
+        len = LLVMBuildZExt(comp_ctx->builder, len, I64_TYPE, "len64");
+        if (!len) {
+            aot_set_last_error("llvm build bitcast failed.");
+            return false;
+        }
+    }
+
     call_aot_memmove = comp_ctx->is_indirect_mode || comp_ctx->is_jit_mode;
     if (call_aot_memmove) {
         LLVMTypeRef param_types[3], ret_type, func_type, func_ptr_type;
         LLVMValueRef func, params[3];
 
         param_types[0] = INT8_PTR_TYPE;
         param_types[1] = INT8_PTR_TYPE;
-        param_types[2] = I32_TYPE;
+        param_types[2] = INTPTR_T_TYPE; /* int32 for 32-bit, int64 for 64-bit */
```

Comment:
Could we perhaps define SIZE_T_TYPE (or MEM_OFFSET_T_TYPE) instead? Even though it will (almost) always have the same value as INTPTR_T_TYPE, semantically pointer and size are different so the code would probably be slightly easier to read and maintain.